src/python_testing/TC_OCC_2_1.py

```diff
@@ -0,0 +1,216 @@
+            TestStep(7, "Read HoldTimeLimits attribute, if supported"),
+            TestStep(8, "Read PIROccupiedToUnoccupiedDelay attribute, if supported"),
+            TestStep(9, "Read PIRUnoccupiedToOccupiedDelay attribute, if supported"),
+            TestStep(10, "Read PIRUnoccupiedToOccupiedThreshold attribute, if supported"),
+            TestStep(11, "Read UltrasonicOccupiedToUnoccupiedDelay attribute, if supported"),
+            TestStep(12, "Read UltrasonicUnoccupiedToOccupiedDelay attribute, if supported"),
+            TestStep(13, "Read UltrasonicUnoccupiedToOccupiedThreshold attribute, if supported"),
+            TestStep(14, "Read PhysicalContactOccupiedToUnoccupiedDelay attribute, if supported"),
+            TestStep(15, "Read PhysicalContactUnoccupiedToOccupiedDelay attribute, if supported"),
+            TestStep(16, "Read PhysicalContactUnoccupiedToOccupiedThreshold attribute, if supported")
+        ]
+        return steps
+
+    def pics_TC_OCC_2_1(self) -> list[str]:
+        pics = [
+            "OCC.S",
+        ]
+        return pics
+
+    @async_test_body
+    async def test_TC_OCC_2_1(self):
+
+        endpoint = self.user_params.get("endpoint", 1)
+
+        self.step(1)
+        attributes = Clusters.OccupancySensing.Attributes
+
+        self.step(2)
+        attribute_list = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.AttributeList)
+
+        self.step(3)
+        if attributes.Occupancy.attribute_id in attribute_list:
+            occupancy_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.Occupancy)
+
+            if occupancy_dut != NullValue:
+                asserts.assert_less_equal(occupancy_dut, 0b00000001, "Occupancy attribute is not in valid range")
+        else:
+            logging.info("Test step skipped")
+
+        self.step(4)
+        if attributes.OccupancySensorType.attribute_id in attribute_list:
+            occupancy_sensor_type_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.OccupancySensorType)
+
+            if occupancy_sensor_type_dut != NullValue:
+                asserts.assert_less(occupancy_sensor_type_dut, Clusters.Objects.OccupancySensing.Enums.OccupancySensorTypeEnum.kUnknownEnumValue,
+                                    "OccupancySensorType is not in valid range")
+        else:
+            logging.info("Test step skipped")
+
+        self.step(5)
+        if attributes.OccupancySensorTypeBitmap.attribute_id in attribute_list:
+            occupancy_sensor_type_bitmap_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.OccupancySensorTypeBitmap)
+
+            if occupancy_sensor_type_bitmap_dut != NullValue:
+                asserts.assert_less_equal(occupancy_sensor_type_bitmap_dut, 0b00000111, "OccupancySensorTypeBitmap attribute is not in valid range")

```

Comment:
suggest pulling this from the generated code so doesn't break is new ones are added later
max = sum(list(Clusters.OccupancySensing.Bitmaps.OccupancySensorTypeBitmap))

Reply:
There won't be any more addition to OccupancySensorTypeBitmap attribute in the future.  This attribute will be deprecated later.